File: app/src/main/java/com/arturogutierrez/openticator/di/component/ActivityComponent.java
package com.arturogutierrez.openticator.di.component;

import android.app.Activity;
import com.arturogutierrez.openticator.di.PerActivity;
import com.arturogutierrez.openticator.di.module.ActivityModule;
import dagger.Component;

@PerActivity
@Component(dependencies = ApplicationComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {
  Activity activity();
}

File: app/src/main/kotlin/com/arturogutierrez/openticator/di/component/ActivityComponent.kt
package com.arturogutierrez.openticator.di.component

import android.app.Activity
import com.arturogutierrez.openticator.di.PerActivity
import com.arturogutierrez.openticator.di.module.ActivityModule
import dagger.Component

@PerActivity
@Component(dependencies = arrayOf(ApplicationComponent::class), modules = arrayOf(ActivityModule::class))
interface ActivityComponent {
  fun activity(): Activity
}
